contracts/CreditVoucher.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.26;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {ERC721Enumerable} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {Pausable} from "@openzeppelin/contracts/utils/Pausable.sol";
+
+import {ICreditVoucher} from "./interfaces/ICreditVoucher.sol";
+
+contract CreditVoucher is ICreditVoucher, ERC721Enumerable, Pausable, EIP712, Ownable {
+    using SafeERC20 for IERC20;
+    using SignatureChecker for address;
+
+    bytes32 public constant VERSION = keccak256("CREDIT_VOUCHER_V1");
+    // keccak256("Mint(address to,uint256 creditValue,uint256 nonce,uint256 deadline)")
+    bytes32 private constant MINT_TYPEHASH = 0xd10efb11fc8417faf3162be9fb4bccfb2fe444b18748b7f8b5693464fbaa56bb;
+
+    address public immutable USDC;
+
+    address public operator;
```

Comment:
what is the difference between operator and owner?

Reply:
The owner has the authority to pause and unpause the contract and can assign a new operator. The operator, on the other hand, can sign lazy minting messages, allowing users to mint vouchers.

Reply:
OK. Could you add a comment about this?